src/cmd_all/src/standalone.rs

```diff
@@ -268,12 +335,23 @@ It SHOULD NEVER be used in benchmarks and production environment!!!"
         );
     }
 
-    // TODO: should we join all handles?
-    // Currently, not all services can be shutdown gracefully, just quit on Ctrl-C now.
-    // TODO(kwannoel): Why can't be shutdown gracefully? Is it that the service just does not
-    // support it?
-    signal::ctrl_c().await.unwrap();
-    tracing::info!("Ctrl+C received, now exiting");
+    // Wait for shutdown signals.
+    tokio::select! {
+        // Meta service stopped itself, typically due to leadership loss of idleness.
+        // Directly exit in this case.
+        _ = meta.as_ref().unwrap().shutdown.cancelled(), if meta.is_some() => {
```

Comment:
Is it okay to just unwrap here?

Reply:
I think so. The precondition is evaluated before creating the future.

> Evaluate all provided `<precondition>` expressions. If the precondition returns false, disable the branch for the remainder of the current call to `select!`. Re-entering `select!` due to a loop clears the “disabled” state.

https://docs.rs/tokio/latest/tokio/macro.select.html

Reply:
You're right. It can be problematic.

> If the branch is disabled, `<async expression>` is still evaluated, but the resulting future is not polled.